libs/core/src/config.ts

```diff
@@ -0,0 +1,38 @@
+import * as dotenv from 'dotenv';
+import { ZodType, z } from 'zod';
+import { Environments } from './ports';
+
+dotenv.config({ path: '../../.env' });
+
+/**
+ * Extends target config with payload after validating schema
+ *
+ * @param target target payload
+ * @param extend extended payload
+ * @param schema extended schema
+ * @returns extended config
+ */
+export const configure = <T, E extends Record<string, unknown>>(
+  target: Readonly<T>,
+  extend: Readonly<E>,
+  schema: ZodType<E>,
+): Readonly<T & E> =>
+  Object.assign(target || {}, schema.parse(extend)) as Readonly<T & E>;
+
+const { NODE_ENV, PORT } = process.env;
+const port = parseInt(PORT || '8080', 10);
+
+export const config = z
+  .object({
+    env: z.enum(Environments),
```

Comment:
Could we maybe maintain the all-caps naming convention so that it's doubly clear that these are environment variables? Also, I don't think we should rename variables e.g. from `NODE_ENV` to `env`. It's confusing and adds additional complexity especially when we have many different `env` variables e.g. `SITEMAP_ENV` or `SERVER_ENV`.

Reply:
I have some ideas about env var naming conventions... let's discuss soon.